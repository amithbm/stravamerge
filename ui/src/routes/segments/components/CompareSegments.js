import React from 'react'
import { Grid, Row, Col } from 'react-bootstrap'
import ActivityListItem from '../../activities/components/ActivityListItem'
import CompareSegmentsTable from './CompareSegmentsTable'
import CompareSegmentsChart from './CompareSegmentsChart'

export default ({activity, segmentData, isLoading }) => isLoading ? <div></div> : <Grid>
    <Row>
        <Col xs={12}>
            <ActivityListItem activity={activity} includeHyperlinks={false} />
        </Col>
    </Row>
    <Row>
        <Col xs={12}>
            <CompareSegmentsChart chartData={segmentData} />
        </Col>
    </Row>
    <Row>
        <Col xs={12}>
            <CompareSegmentsTable tableData={segmentData} />
        </Col>
    </Row>
</Grid>
