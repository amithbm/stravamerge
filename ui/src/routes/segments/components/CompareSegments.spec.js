import React from 'react'
import {shallow} from 'enzyme'
import CompareSegments from './CompareSegments'
import toJson from 'enzyme-to-json'

describe('components', () => {
    describe('<CompareSegments />', () => {
        it('renders content loaded', () => {
            const component =  shallow(<CompareSegments isLoading={false} segmentData={[]} activity={{}} />)
            const tree = toJson(component)
            expect(tree).toMatchSnapshot()
        })

        it('renders empty when not loaded', () => {
            const component =  shallow(<CompareSegments isLoading={true} />)
            const tree = toJson(component)
            expect(tree).toMatchSnapshot()
        })
    })
})

