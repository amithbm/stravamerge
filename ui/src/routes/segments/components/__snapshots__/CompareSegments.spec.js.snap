exports[`components <CompareSegments /> renders content loaded 1`] = `
<Grid
  bsClass="container"
  componentClass="div"
  fluid={false}>
  <Row
    bsClass="row"
    componentClass="div">
    <Col
      bsClass="col"
      componentClass="div"
      xs={12}>
      <Component
        activity={Object {}}
        includeHyperlinks={false} />
    </Col>
  </Row>
  <Row
    bsClass="row"
    componentClass="div">
    <Col
      bsClass="col"
      componentClass="div"
      xs={12}>
      <CompareSegmentsChart
        chartData={Array []} />
    </Col>
  </Row>
  <Row
    bsClass="row"
    componentClass="div">
    <Col
      bsClass="col"
      componentClass="div"
      xs={12}>
      <CompareSegmentsTable
        tableData={Array []} />
    </Col>
  </Row>
</Grid>
`;

exports[`components <CompareSegments /> renders empty when not loaded 1`] = `<div />`;
